tools/third_party_modified/mozlog/mozlog/formatters/grouping.py

```diff
@@ -150,7 +151,7 @@ def suite_start(self, data):
             )
 
     def test_start(self, data):
-        self.running_tests[data["thread"]] = data["test"]
+        self.running_tests[data["thread"]] = (data["subsuite"], data["test"])
```

Comment:
The RFC says the [`subsuite` key is optional](https://github.com/web-platform-tests/rfcs/blob/51538dcf1666eb2a4e6c52386bb49afe5441b3af/rfcs/subsuites.md?plain=1#L163). Therefore, should this (and elsewhere) be `data.get("subsuite")` (if you choose `None` to represent the default subsuite)? (Should also help the tests pass.)

Reply:
Looking at the implementation in https://github.com/web-platform-tests/wpt/pull/39711, `tests_to_run` etc is keyed with `(subsuite_name, test_name)`, where subsuite_name is the empty string for generic suite. So the subsuite_name is mandatory even though we want it to be optional in the RFC.

There once had a discussion if we want to use the empty string or None to represent the default subsuite. Even though we (chromium) to prefer to use None, jgraham@ choose to use the empty string. There might have a technical reason behind this. For example it might be difficult to represent None in metadata file. I think we can do a follow up change in future to use None instead of the empty string.

Yes using `data.get("subsuite")` can help pass the unit tests. I will take a look to see if we should use `data.get("subsuite")` and add unit tests to test the new scenarios, or make change to existing unit tests.

Reply:
> So the subsuite_name is mandatory even though we want it to be optional in the RFC.

The example of `test_to_run` in #39711 is different, since that's an internal wptrunner data structure. OTOH, if we ever choose to reconcile this vendored `mozlog` version with the `mozbase` one, the formatters should gracefully accommodate callers that don't pass `subsuite`.

> I think we can do a follow up change in future to use None instead of the empty string.

I agree that `None` would be preferable to make the default subsuite's special status obvious.